server/tserver/src/main/java/org/apache/accumulo/tserver/compactions/Compactable.java

```diff
@@ -68,7 +68,7 @@ public Files(SortedMap<StoredTabletFile,DataFileValue> allFiles,
       this.candidates = Collections.unmodifiableSet(candidates.stream()
           .map(stf -> new CompactableFileImpl(stf, allFiles.get(stf))).collect(Collectors.toSet()));
 
-      this.compacting = Set.copyOf(running);
+      this.compacting = runningJobsCopy;
```

Comment:
I think its ok to leave this as `this.compacting = Set.copyOf(running);`. I just looked at the impl of Set.copyOf and it looks like the following.

```java
    static <E> Set<E> copyOf(Collection<? extends E> coll) {
        if (coll instanceof ImmutableCollections.AbstractImmutableSet) {
            return (Set<E>)coll;
        } else {
            return (Set<E>)Set.of(new HashSet<>(coll).toArray());
        }
    }
```

So if it when its something created by Set.of or Set.copyOf, is seems copyOf will not copy.   Leaving the copyOf call in the code currently avoids a second copy while being robust to changes outside the constructors.

Reply:
Oh OK cool. Thanks for looking into the impl.